File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] ans = new int[nums.length];
        int start=0;
        int end=nums.length-1;
        int index=nums.length-1;

        while(start<=end){
            int leftsq=nums[start]*nums[start];
            int rightsq=nums[end]*nums[end];
            if(leftsq>rightsq){
                ans[index]=leftsq;
                index--;
                start++;
            }
            else{
                ans[index]=rightsq;
                index--;
                end--;
            }

        }
        return ans;
    }
}

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int start=0;
        int end= nums.size()-1;
        vector<int> result(nums.size());
        int index = nums.size()-1;

        while(start<=end){
            int first=nums[start]*nums[start];
            int second=nums[end]*nums[end];
            if(first>second){
                result[index]=first;
                start++;
                index--;
            }
            else{
                result[index]=second;
                end--;
                index--;
            }
        }
        return result;
    }
};